message/infrastructure/http/publisher.go

```diff
@@ -0,0 +1,161 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+var (
+	// ErrPublisherClosed happens when trying to publish to a topic while the publisher is closed or closing.
+	ErrPublisherClosed = errors.New("publisher is closed")
+	ErrNoMarshalFunc   = errors.New("marshal function is missing")
+
+	ErrErrorResponse = errors.New("server responded with error status")
+)
+
+type MarshalMessageFunc func(topic string, msg *message.Message) (*http.Request, error)
+
+// DefaultMarshalMessageFunc returns a MarshalMessage func transforming the message into a HTTP POST request.
+// It encodes the UUID and Metadata in request headers.
+// The request URL is combined from the base address and the topic.
+func DefaultMarshalMessageFunc(address string) MarshalMessageFunc {
+	return func(topic string, msg *message.Message) (*http.Request, error) {
+		req, err := http.NewRequest(http.MethodPost, address+"/"+topic, bytes.NewBuffer(msg.Payload))
```

Comment:
what about just changing topic to entire url?

Reply:
b97952c46ffe87d5349ecb9ef5a191baa72fbd80